tests/llms/test_openai.py

```diff
@@ -69,6 +72,40 @@ def mock_completion_stream(*args: Any, **kwargs: Any) -> Generator[dict, None, N
         yield response
 
 
+class CachedOpenAIApiKeys:
+    """
+    Saves the users' OpenAI API key either in the environment variable
+    or set to the library itself.
+    This allows us to run tests by setting it without plowing over
+    the local environment.
+    """
```

Comment:
Although I don't think this is necessary, due to comments below, it did give me the idea that we can make our users' lives easier by storing their API keys in llama cache dir. Any thoughts @logan-markewich

Reply:
One thing I like about having to provide API keys is that it stops me from accidentally running some expensive API calls when what I intended was to use some local models...

Maybe something like 

```python
from llama_index import enable_cached_api_keys
enable_cached_api_keys()
```

Reply:
I think we can leave the key management in the users hands for now :)